java/ql/src/experimental/Security/CWE/CWE-208/TimingAttackAgainstHeader.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Timing attack against headers value
+ * @description A constant-time algorithm should be used for checking the value of headers. 
+ *              In other words, the comparison time should not depend on the content of the input
+ *              Otherwise, an attacker may be able to implement a timing attacks that may reveal the value of sensitive headers
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/timing-attack-against-headers-value
+ * @tags security
+ *       external/cwe/cwe-208
+ */
+
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import DataFlow::PathGraph
+
+private class NonConstantTimeEqualsCall extends MethodAccess {
+  NonConstantTimeEqualsCall() {
+    this.getMethod().hasQualifiedName("java.lang", "String", ["equals", "contentEquals", "equalsIgnoreCase"]) or
+    this.getMethod().hasQualifiedName("java.nio", "ByteBuffer", ["equals", "compareTo"])
+  }
+}
+
+private predicate isNonConstantTimeEqualsCall(Expr firstObject, Expr secondObject) {
+  exists(NonConstantTimeEqualsCall call |
+    firstObject = call.getQualifier() and
+    secondObject = call.getAnArgument()
+    or
+    firstObject = call.getAnArgument() and
+    secondObject = call.getQualifier()
+  )
```

Comment:
If you're always expecting argument 0, use `getArgument(0)` instead of `getAnArgument()`.

If the expected position of the thing being compared varies from method to method, check which method is being called, otherwise you'll produce false positives regarding the method's other arguments.